config/systempatches/patches.inc

```diff
@@ -200,36 +195,38 @@ function bootup_apply_patches() {
 function patch_add_shellcmd() {
 	global $config;
 	$a_earlyshellcmd = &$config['system']['earlyshellcmd'];
-	if (!is_array($a_earlyshellcmd)) {
-		$a_earlyshellcmd = array();
-	}
 	$found = false;
-	foreach ($a_earlyshellcmd as $idx => $cmd) {
-		if (stristr($cmd, "apply_patches.php")) {
-			$found = true;
+	if (is_array($a_earlyshellcmd)) {
+		foreach ($a_earlyshellcmd as $idx => $cmd) {
+			if (stristr($cmd, "apply_patches.php")) {
+				$found = true;
+			}
 		}
+	} else {
+		$a_earlyshellcmd = array();
```

Comment:
Surely not. The one below, yes but this one adds an item to earlyshellcmd if nothing is found, and will therefore always need the array created if it didn't exist, since that's the most obvious way for !$found to be true